morpheus/_lib/src/messages/multi_inference.cpp

```diff
@@ -52,12 +52,12 @@ const TensorObject MultiInferenceMessage::get_input(const std::string &name) con
     // check if we are getting the entire input
     if (this->offset == 0 && this->count == this->memory->count)
     {
-        return this->memory->inputs[name];
+        return this->memory->tensors[name];
     }
 
     // TODO(MDD): This really needs to return the slice of the tensor
```

Comment:
It's something that grew out of this issue. Since the `InferenceMemory` and `ResponseMemory` classes were 99% the same code, it made sense to create a common base-class `TensorMemory` to avoid repeating the code contained in the `copy_tensor_ranges` method.